crates/editor/src/editor.rs

```diff
@@ -2288,22 +2289,32 @@ impl Editor {
         key_context
     }
 
-    fn show_mouse_cursor(&mut self) {
-        self.mouse_cursor_hidden = false;
+    fn show_mouse_cursor(&mut self, cx: &mut Context<Self>) {
+        self.set_hide_mouse_cursor(false, cx)
     }
 
-    pub fn hide_mouse_cursor(&mut self, origin: &HideMouseCursorOrigin) {
-        self.mouse_cursor_hidden = match origin {
-            HideMouseCursorOrigin::TypingAction => {
-                matches!(
-                    self.hide_mouse_mode,
-                    HideMouseMode::OnTyping | HideMouseMode::OnTypingAndMovement
-                )
-            }
-            HideMouseCursorOrigin::MovementAction => {
-                matches!(self.hide_mouse_mode, HideMouseMode::OnTypingAndMovement)
-            }
-        };
+    pub fn hide_mouse_cursor(&mut self, origin: HideMouseCursorOrigin, cx: &mut Context<Self>) {
+        self.set_hide_mouse_cursor(
+            match origin {
+                HideMouseCursorOrigin::TypingAction => {
+                    matches!(
+                        self.hide_mouse_mode,
+                        HideMouseMode::OnTyping | HideMouseMode::OnTypingAndMovement
+                    )
+                }
+                HideMouseCursorOrigin::MovementAction => {
+                    matches!(self.hide_mouse_mode, HideMouseMode::OnTypingAndMovement)
+                }
+            },
+            cx,
+        );
+    }
+
+    fn set_hide_mouse_cursor(&mut self, hide_mouse_cursor: bool, cx: &mut Context<Self>) {
+        if self.mouse_cursor_hidden != hide_mouse_cursor {
+            self.mouse_cursor_hidden = hide_mouse_cursor;
+            cx.notify();
+        }
```

Comment:
I think it’s fine to move the logic to `hide_mouse_cursor` and `show_mouse_cursor`. A bit repetition is okay, as `set_hide_mouse_cursor` is confusing since we already have two methods for similar to that.